test/addons-napi/test_error/test.js

```diff
@@ -55,3 +55,37 @@ assert.strictEqual(test_error.checkError({}), false,
 // Test that non-error primitive is correctly classed
 assert.strictEqual(test_error.checkError('non-object'), false,
                    'Non-error primitive correctly classed by napi_is_error');
+
+assert.throws(() => {
+  test_error.throwExistingError();
+}, /Error: existing error/);
```

Comment:
We've been moving toward adding `^` and `$` to make the regular expression stricter.